src/deepsparse/transformers/utils/decoder_kv_cache.py

```diff
@@ -241,7 +238,11 @@ def num_non_blank_entries(self):
         """
         :return: the number of non-blank entries in the kv cache
         """
-        return min(self.capacity, self.total_num_processed_tokens)
+
+        # given a 2D array get number of rows that are not entirely zero
+        single_kv_cache_entry = self.cached_inputs[list(self.cached_inputs.keys())[0]]
```

Comment:
why are we calculating this indirectly instead of just tracking the number of total tokens for the session as an attribute

Reply:
additionally - let's make sure calculating positions correctly - it shouldn't have anything with blank entries or padding, just the number of real tokens